src/main/perl/ComponentProxyList.pm

```diff
@@ -146,31 +172,42 @@ sub run_all_components
             $self->set_state($name, "");
 
             my $ret = $comp->executeConfigure();
-            if (!defined($ret)) {
-                my $err = "cannot execute configure on component " . $name;
-                $self->error($err);
-                $status->{'ERRORS'}++;
-                $status->{ERR_COMPS}->{$name} = 1;
-                $self->set_state($name, $err);
-            } else {
+            if (defined($ret)) {
+                # errors before warnings (errors can be downgraded to warnings)
                 if ($ret->{'ERRORS'}) {
-                    $status->{ERR_COMPS}->{$name} = $ret->{ERRORS};
-                    $self->set_state($name, $ret->{ERRORS});
+                     if ($nodepsnoerrors) {
+                         # Convert errors in warnings
+                         # TODO do we set the state to failed?
+                         $self->warn("Errors from $name are downgraded to warnings (nodepsnoerrors is set). ",
+                                     "State is not cleared (as something went wrong), but also not set.");
```

Comment:
Not sure what this line means... And the previous line looks more as a debug message than a warning message. What about a warning text like "$name execution failure ignored (nodepsnoerror is set)" and possibly a debug message with more information if neeeded.